### ISSUE TITLE
Compatibility with Elastic 8

### ISSUE THREAD

**Issue by AlexRuiz7** [wazuh/wazuh-kibana-app#4567]
## Description
We need to ensure the UI compatibility with the next version of Elastic 8.
This update is still being discussed, but we need to be aware of potential issues.

For that, we need to:

- [x] Review Elastic 8 and Kibana 8 latest stable changelog.
- [x] Identify improvements and potential impact on the UI.
- [x] Develop a testing environment to verify our components would work under this new build.

For each of the following versions:

- 8.7.0 (no released yet)
- 8.5.3
- 8.5.2
- 8.5.1
- [x] **8.5.0**
- 8.4.3
- [x] **8.4.2**
- 8.4.1
- 8.4.0
- [x] **8.3.3**
- 8.3.2
- 8.3.1
- 8.3.0
- [x] **8.2.3**
- 8.2.2
- 8.2.1
- 8.2.0
- [x] **8.1.3**
- 8.1.2
- 8.1.1
- 8.1.0
- 8.0.1
- 8.0.0

We'll focus in testing the latest patch version of each minor, these in bold. We'll skip the remaining versions.

## Issues
-  _List here the detected issues_

## Documentation
- Migration to Kibana 8.0 https://github.com/elastic/kibana/blob/v8.5.0/docs/migration/migrate_8_0.asciidoc

**Comment by chantal-kelm** [comment]
## Elastic 8.0
 
### [Breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking-changes-8.0)

- [Cluster and node setting changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_cluster_node_setting_changes)
- [Command line tool changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_command_line_tool_changes)
- [Index settings address](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_index_setting_changes)
- [Java API changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_java_api_changes)
- [JVM options change](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_jvm_option_changes)
- [Logging changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_logging_changes)
- [Mapping changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_mapping_changes)
- [Packaging changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_packaging_changes)
- [Painless changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_painless_changes)
- [Plugin changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_plugin_changes)
- [Rest API changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_rest_api_changes)
- [SQL JDBC changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_jdbc_changes)
- [System req changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_system_req_changes)
- [Transform Changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_transform_changes)

### [Deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#deprecated-8.0)

- [Cluster and node setting deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_cluster_node_setting_deprecations)
- [Command line tool deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_command_line_tool_deprecations)

**Comment by chantal-kelm** [comment]
## Elastic 8.1
 
### [Breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.1.html#breaking-changes-8.1)

- [Rest API changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.1.html#breaking_81_rest_api_changes)

### [Deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#deprecated-8.0)

> The following functionality has been deprecated in Elasticsearch 8.1 and will be removed in a future version. While this won’t have an immediate impact on your applications, we strongly encourage you take the described steps to update your code after upgrading to 8.1.

- [Cluster and node setting deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.1.html#deprecations_81_cluster_and_node_setting)

> Legacy values for the discovery.type setting are deprecated.

- [REST API deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.1.html#deprecations_81_rest_api)

> Lenient parsing of bulk actions is deprecated.

> Deprecate index_include_frozen request parameter in _sql API.

**Comment by chantal-kelm** [comment]
## Elastic 8.2
 
### [Breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.2.html#breaking-changes-8.2)

> There are no breaking changes in Elasticsearch 8.2.

**Comment by chantal-kelm** [comment]
## Elastic 8.3
 
### [Breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.3.html#breaking-changes-8.3)

> There are no breaking changes in Elasticsearch 8.3.

### [Deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.3.html#deprecated-8.3)

> The following functionality has been deprecated in Elasticsearch 8.3 and will be removed in a future version. While this won’t have an immediate impact on your applications, we strongly encourage you take the described steps to update your code after upgrading to 8.3.

- [Cluster and node setting deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.3.html#deprecations_83_cluster_and_node_setting)

> Configuring a bind DN in an LDAP or Active Directory (AD) realm without a corresponding bind password is deprecated

**Comment by chantal-kelm** [comment]
## Elastic 8.4
 
### [Breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.4.html#breaking-changes-8.4)

> There are no breaking changes in Elasticsearch 8.3.

### [Deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.4.html#deprecated-8.4)

> The following functionality has been deprecated in Elasticsearch 8.4 and will be removed in a future version. While this won’t have an immediate impact on your applications, we strongly encourage you to take the described steps to update your code after upgrading to 8.4.

- [REST API deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.4.html#deprecations_84_rest_api)

> Deprecate the _knn_search endpoint

**Comment by chantal-kelm** [comment]
## Elastic 8.5
 
### [Breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.5.html#breaking-changes-8.5)

- [REST API changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.5.html#breaking_85_rest_api_changes)

> The bulk API now rejects requests containing unrecognized actions

### [Deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.5.html#deprecated-8.5)

> The following functionality has been deprecated in Elasticsearch 8.5 and will be removed in a future version. While this won’t have an immediate impact on your applications, we strongly encourage you to take the described steps to update your code after upgrading to 8.5.

- [Plugin API deprecations](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.5.html#deprecations_85_plugins)

> Plugins that extend the NetworkPlugin interface are deprecated.

**Comment by chantal-kelm** [comment]
## Elastic 8.6
 
### [Breaking changes](https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.6.html#breaking-changes-8.6)

> There are no breaking changes in Elasticsearch 8.6.

**Comment by yenienserrano** [comment]
Testing to run kibana 8.4.3 with integrated wazuh.

Change the imports from @kbn/i18n/react to @kbn/i18n-react in the files:

- registry-table.tsx
- kbn-search-bar.tsx
- no_results.js
- uninitalized.tsx
- tool_bar_pager_text.tsx
- discover_legacy.tsx
- context_error_message.tsx
- doc.tsx
- doc_viewer_tab.tsx
- hits_counter.tsx
- loading_spinner.tsx
- discover_field_details.tsx
- discover_field_search.tsx
- discover_index_pattern_title.tsx
- discover_index_pattern.tsx
- discover_sidebar.tsx
- skip_bottom_button.tsx
- table_row_btn_filter_add.tsx
- table_row_btn_filter_exists.tsx
- table_row_btn_filter_remove.tsx
- table_row_btn_toggle_column.tsx
- timechart_header.tsx
- open_search_panel.js

Problems with `kibana_legacy` plugin that was removed since version 8.1.0.
Copy this plugin to the kibana integrations folder and modify the paths where it was being used.
At this point kibana is up but the wazuh app is not visible and there are no errors in the kibana logs.

As far as I can see, in [Kibana documentation](https://www.elastic.co/guide/en/kibana/current/migrating-legacy-plugins.html), it is said that you have to migrate from kibana_legacy to kibanaPlataform.

This migration from what they say is not an easy task, they give some steps to follow but depending on each application you have to make more or less changes.
This would mean having to create another branch in the repository to be compatible with 8.x versions from what I understand, which would mean another branch to maintain.

**Comment by Desvelao** [comment]
# Update
I was researching and I got the mentioned problems.

I could solve some of these problems:
- Moved dependency: `@kbn/i18n/react` to `@kbn/i18n-react` 
- Missing files in `src/plugins/kibana_legacy`. In the plugin, we have the methods imported in some files. It only requires import the plugin files instead of the plugin platform.

After these fixes, the plugin doesn't throw errors when transpiling the code. But, it doesn't work.

In my first view, there are other required changes:
- Remove the `kibanaLegacy` dependency from the plugin platform manifest file `kibana.json`
- In the frontend, it is required to register the plugin in the `setup` function. Currently, we register the plugin after a request. We should move the request and doesn't block the plugin registration. I got the plugin that appears in the plugin platform menu.
![image](https://user-images.githubusercontent.com/34042064/199532364-ad02d832-7566-4fa7-a594-e78cd1376a39.png)
- The information about the index used by the platform seems to be removed. This is affecting to the `initialize` job of the plugin on the backend.
- I got some errors related to the `security` client that the plugin adds under the `wazuh` route `context`. I could not find a solution.

Unfortunately, these errors are the first ones and we can't see any more errors until we solve them. We have to solve each error until we don't get any errors. After solving these problems, we have to test the UI and each functionality (monitoring, statistics, etc...)

**Comment by Desvelao** [comment]
# Problems

| Error |  Comment/Solution | Solved |
| --- | --- | --- |
| Renamed dependency `@kbn/i18n/react` | Rename to `@kbn/i18n-react` | :heavy_check_mark: |
| Imports from `src/plugins/kibana_legacy` | Use the dependencies hosted in the plugin itself | :heavy_check_mark: |
| Removed plugin `kibanaLegacy` | Remove dependency of the plugin manifest `kibana.yml` | :heavy_check_mark: |
| Can't get the platform index | The configuration object doesn't include the index | :x: |
| Can't import `SearchBar` component | The component was moved to the `unifiedSearch` plugin | :x: |

**Comment by carpetmaker** [comment]
What is the current (as well as any future plans) status of Wazuh Kibana plugin compatibility with [OpenSearch Dashboards (Kibana fork; ex OpenDistro)](https://github.com/opensearch-project/OpenSearch-Dashboards)?
Wazuh Indexer is already based on the more recent OpenSearch (ElasticSearch fork; ex OpenDistro) versions...
[This is their own changelog](https://opensearch.org/docs/2.0/version-history/), however they don't explicitly state which version is based on which Kibana OSS version...

**Comment by AlexRuiz7** [comment]
@carpetmaker [Wazuh Dashboard](https://documentation.wazuh.com/current/getting-started/components/wazuh-dashboard.html) is based on OpenSearch Dashboards and already includes a compatible plugin.

4.3.x versions are based in OpenSearch 1.2.0.
4.4.x versions will be based in OpenSearch 2.3.0, as described here: #4160 

We do not plan to provide a standalone plugin for OpenSearch Dashboards, as we already provide a built-in plugin for Wazuh Dashboard.

Btw, OpenDistro (and later on OpenSearch) was forked from Elastic 7.10.2

**Comment by Desvelao** [comment]
We are going to work using as base the last current version of Kibana: `8.5.0`.

We will create a development branch where we will implement the solutions to the compatibility problems.

**Compatibility issues**

| Error | Comment/Solution | Solved (include comit) |
| --- | ---- | --- |
| Renamed dependency `@kbn/i18n/react` | Rename to `@kbn/i18n-react` |  |
| Renamed dependency `@elastic/safer-lodash-set` | Rename to `@kbn/safer-lodash-set` |  |
| Error creating a template for .kibana | |  |
| Imports from `src/plugins/kibana_legacy` | Use the dependencies hosted in the plugin itself |  |
| Removed plugin `kibanaLegacy` | Remove dependency of the plugin manifest `kibana.json` |  |
| Plugin lifecycle methods should be synchronous | Adapt to synchronous |  |
| The plugin on frontend should be registered synchronous | Adapt to synchronous  | |
| Can't get the platform index name | The configuration object doesn't include the index | |
| Can't import `SearchBar` component | The component was moved to the `unifiedSearch` plugin |  |
| Initialize (check plugin platform index existence) and any responses of Elasticsearch client | The response of the client doesn't  come from `body` property | :x: |
| `SavedObjectLoader` is not exported from the `saved_object` plugin | |  |
| The `context` in the endpoints is a promise |  |  |
| Cannot read properties of undefined (reading 'FilterStateStore') | Delete the `esFilters.` and import directly from `@kbn/es-query` |  |
| visualizations.createVis is not a function |  |  |

**Comment by Desvelao** [comment]
# Get the platform index name

The platform configuration `kibana.index` was removed in Kibana 8.0.0-beta1 https://www.elastic.co/guide/en/kibana/8.0/release-notes-8.0.0-beta1.html#breaking-112773.

I was reviewing the related issues and PRs:
- https://github.com/elastic/kibana/issues/82020
- https://github.com/elastic/kibana/issues/82521
- https://github.com/elastic/kibana/pull/112773

and I found a way to get this index. This can be obtained from the `core.savedObjects.getKibanaIndex()` (https://github.com/elastic/kibana/pull/112773/files#diff-518c24f824853d21141674f3bfbb9215fe6516f6086775ed2b2e41eef636e0a2R200)

**Comment by Desvelao** [comment]
# Applications cannot be registered after "setup"

The plugin setup should be synchronous. In the plugins of previous Kibana versions, before the plugin registration, we do a request to get the customized logos and register the plugin with the customized logos, it avoids the logos change on the UI. 

I added a commit to register the plugin using the default images, and after, when the plugin is mounted on the UI, could change the logos causing a blink.

**Tasks**
- [ ] Review if the customization of logos (menu and sidebar) cause some blink on the UI.

**Comment by Desvelao** [comment]
# visualizations.createVis is not a function

The `createVis` function is not available in objects exported by the `visualizations` plugin in the plugin lifecycle.

In 8.5.0: https://github.com/elastic/kibana/blob/v8.5.0/src/plugins/visualizations/public/plugin.ts#L390-L393
In 7.17.5:: https://github.com/elastic/kibana/blob/v7.17.5/src/plugins/visualizations/public/plugin.ts#L235-L236

References:
- https://github.com/elastic/kibana/pull/121550

# Digging
I was researching and in Kibana 7.17.5, the `createVis` function is a wrapper of another function called `createVisAsync` https://github.com/elastic/kibana/blob/v7.17.5/src/plugins/visualizations/public/plugin.ts#L235-L236. 

In Kibana 8.5.0, the `createVisAsync` continues in the same file: https://github.com/elastic/kibana/blob/v8.5.0/src/plugins/visualizations/public/vis_async.ts#L12-L25.

I reviewed the stuff exported by the `visualizations` plugin and this doesn't export directly the `createVisAsync` function https://github.com/elastic/kibana/blob/v8.5.0/src/plugins/visualizations/public/index.ts.

Currently, I am researching if another exported method is a wrapper of this function.

**Comment by yenienserrano** [comment]
# Object(...)(...).convertToSerializedVis is not a function

The convertToSerializedVis function is not available in objects exported by the visualizations plugin in the plugin lifecycle.

in 8.5.0: https://github.com/elastic/kibana/blob/v8.5.0/src/plugins/visualizations/public/utils/saved_visualize_utils.ts#L139-L154
in 7.17.5: https://github.com/elastic/kibana/blob/v7.17.5/src/plugins/visualizations/public/plugin.ts#L237

**Comment by yenienserrano** [comment]
# Cannot read properties of undefined (reading 'id') when the menu opens

I was investigating the error, but it seems to be related to the indexes and their fields.

so I think it is related to the problem of adding sample data since it is about indexes, I am investigating to see if I can fix that sample data error and see if this error is fixed, since as far as I can see it comes from a problem reading the indexes.

**Comment by yenienserrano** [comment]
Error in the default filters of the views, warning appears instead of the filter

![image](https://user-images.githubusercontent.com/63758389/203273892-370ee257-d3f1-4efd-bc6a-d7b4fd5e2d8a.png)

**Comment by yenienserrano** [comment]
with the indexes working there is no longer a problem with filters

![image](https://user-images.githubusercontent.com/63758389/203601001-9b5408a0-effb-412e-89a9-0f6bd91cce19.png)

**Comment by Desvelao** [comment]
# FieldIcon, FieldButton components are missing

These components were imported from the `kibana_react` plugin. https://github.com/elastic/kibana/blob/v7.17.6/src/plugins/kibana_react/public/index.ts#L19-L20

In Kibana 8.5.0, they are not available there: https://github.com/elastic/kibana/blob/v8.5.0/src/plugins/kibana_react/public/index.ts. It seems they were moved to `@kbn/react-field` https://github.com/elastic/kibana/blob/v8.5.0/packages/kbn-react-field/index.ts

**Comment by Desvelao** [comment]
# The table row of results in the Events tabs is not rendered

If found a problem with the creation of the angular directive:

![Image](https://user-images.githubusercontent.com/34042064/205104766-5c779581-c904-4622-a829-257ac89d6977.png)

## Digging

I was researching and this could be related to the CSP (Content Security Policy) of the page. This is set to disallow the eval of JavaScript code.

Comparison in:

- Wazuh dashboard 4.3.x
![Image](https://user-images.githubusercontent.com/34042064/205105639-d72db1b7-2d2b-426e-8aba-b3c7c36cea60.png)

- Kibana 8.5.0 (dev mode)
![Image](https://user-images.githubusercontent.com/34042064/205105289-e23b1567-f6b9-4542-99c4-994f09b77c64.png)

The error is coming from the `template` method of lodash. This function evals code and the CSP (Content security policy) disallow it, so the AngularJS directive can't be set. We should search for an alternative to workaround this problem.

**Comment by AlexRuiz7** [comment]
Elastic has releases v8.5.2, v8.5.3 and a v8.7.0 is on the way.
https://www.elastic.co/guide/en/elasticsearch/reference/master/es-release-notes.html

**Comment by Desvelao** [comment]
# Compatibility current status

There are 2 big things that need much effort:
- **Adapt the building of the visualizations related to Elasticsearch data**
The previous way to build the visualizations is not compatible with Kibana 8.5.x. We researched how to render these visualizations and did a POC with some of them where we saw that this could be the way to create them. There are some unsolved problems as the connection of the search bar with the visualizations and viceversa that should solve to ensure the same functionality. More information: https://github.com/wazuh/wazuh-kibana-app/issues/4567#issuecomment-1307409214. Adapting the visualizations and dashboard could require a significant amount of time.

- **Fix the problems in the `Events` tab of the modules**
We identified some problems and saw the possible solutions for some of them, but there are others that are not solved. We can not check if there are more problems until we solve the previous problems.

:exclamation:  **It is important to know that after solving the mentioned problems, others could appear when exploring other sections or features of the Wazuh plugin.** Unfortunately, some of the unsolved problems block us to check other plugin functionalities, for this reason, we need to resolve them before proceeding.

**Comment by florian-besser** [comment]
Hi @Desvelao 

We have faced the same issue, but we're on elastic `8.5.1`.
We've built a custom kibana plugin using your guide at https://documentation.wazuh.com/current/development/packaging/generate-wazuh-kibana-app.html with
```
./generate_wazuh_app.sh -b 4.4-8.5-dev -s ./wazuhapp/kibana/out -r 1
```
We did a quick tweak to let the plugin work in Kibana `8.5.1` and it seems the plugin is installed successfully.

Unfortunately we receive a `405 method not allowed` error when pointing the plugin to our Wazuh server, despite having correct credentials:
```
kibana@kibana-kibana-6b98dbff7-jcndz:~$ curl -u wazuh-wui:wazuh-wui -k "https://wazuh:55000/security/user/authenticate?raw=true"
eyJhbGciOiJFUzUxMiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ3YXp1aCIsImF1ZCI6IldhenVoIEFQSSBSRVNUIiwibmJmIjoxNjczNTgxODQ5LCJleHAiOjE2NzM1ODI3NDksInN1YiI6IndhenVoLXd1aSIsInJ1bl9hcyI6ZmFsc2UsInJiYWNfcm9sZXMiOlsxXSwicmJhY19tb2RlIjoid2hpdGUifQ.ATbBEiOu--nqQhWYRkAY31EhRLMxxDiifzD3Li_7_xX3iHH5Inpp8VRqn6gnDYV_xN1bfd-AwKcUfWVT_WXwH6zrAdaEyxuIiRqkwzZdThmGgsDFvrN21F3AoFEOsHxlX1je3e6-QwniOWnOArRwMm-Nu59r_SygluglOZhsP-mdnmAc
```

I reckon this is because the Kibana plugin expects Wazuh `4.4` and not Wazuh `4.3` that we have installed.

I would appreciate if Wazuh could provide guidance how to install with the current versions of the ELK stack **or provide explicit guidance that the endeavor is not going to work.**
Basically getting to the current point took me a few days and I'm now stuck between half-working components and unsure if I should invest time to try and fix the rest or just abandon the work.

From the top post it seems elastic `8.5.0` was tested successfully, which gave me hope that the Kibana plugin could be built and deployed successfully, given that `8.5.1` hasn't introduced any breaking changes that I'm aware of.

Maybe you have a guide somewhere that includes a compatible wazuh server with a kibana plugin compiled against ELK 8.5.x?

**Comment by Desvelao** [comment]
Hi @florian-besser,

What branch did you use to build the package?

The method to authenticate in Wazuh API 4.4.0 will be unified to `POST` instead of using `GET` for non-context authentication.  In this issue, we were working in a plugin compatible with Wazuh API 4.4.0 and Kibana 8.5.0 (`4.4-8.5-dev`) so if you built a plugin package from this branch, then the plugin expects a Wazuh API 4.4.0. If you are using a previous Wazuh API version, you could change the request to authenticate.

Comparison:
Wazuh 4.3.10 and Kibana 7.17.6: https://github.com/wazuh/wazuh-kibana-app/blob/v4.3.10-7.17.6/server/lib/api-interceptor.ts#L46-L71
Development branch related to this issue: https://github.com/wazuh/wazuh-kibana-app/blob/4.4-8.5-dev/server/lib/api-interceptor.ts#L46-L71

This pull request (https://github.com/wazuh/wazuh-kibana-app/pull/4103) changed the authentication for plugin compatibles with Wazuh API 4.4.0. These changes are included in the development branch related to this issue.

**You should know that there is no plugin compatible with Kibana 8.x**

In this issue, we tried previous versions of the Wazuh plugin in Kibana 8.x. We detected some compatibility problems that need fixes so we were working in a development branch `4.4-8.5-dev` pointing to Wazuh API 4.4.0 and Kibana 8.5.0. Unfortunately, at the moment the issue is on hold, and there are unsolved problems in that branch.

**Comment by florian-besser** [comment]
I used `4.4-8.5-dev` as you mentioned, yes.

After trying to build wazuh `4.4` from source to fulfill this requirement (and that for some reason stalling on our CI...) I stopped trying to force wazuh to use our existing ES8 stack and followed https://documentation.wazuh.com/current/deployment-options/deploying-with-kubernetes/kubernetes-deployment.html to recreate a (slightly duplicate) stack that has compatible versions.

Not my first choice as we're now using a bit too much in terms of resources - but hey, I got something running.

So maybe if someone else sees this thread: Stick to what works (and specifically what is documented to work!) or then be ready to start developing / contributing.

**Comment by sqrtZeroKnowledge** [comment]
> ## Update
> 
> I was researching and I got the mentioned problems.
> 
> > My research was based on Kibana 8.4.2
> 
> I could solve some of these problems:
> 
>     * Moved dependency: `@kbn/i18n/react` to `@kbn/i18n-react`
> 
>     * Missing files in `src/plugins/kibana_legacy`. In the plugin, we have the methods imported in some files. It only requires to import the plugin files instead of the plugin platform.
> 
> 
> After these fixes, the plugin doesn't throw errors when transpiling the code. But, it doesn't work.
> 
> In my first view, there are other required changes:
> 
>     * Remove the `kibanaLegacy` dependency from the plugin platform manifest file `kibana.json`
> 
>     * In the frontend, it is required to register the plugin in the `setup` function. Currently, we register the plugin after a request. We should move the request and doesn't block the plugin registration. I got the plugin that appears in the plugin platform menu, but **this is not visible, there are more problems to solve**.
>       ![image](https://user-images.githubusercontent.com/34042064/199532364-ad02d832-7566-4fa7-a594-e78cd1376a39.png)
> 
>     * The information about the index used by the platform seems to be removed. This is affecting to the `initialize` job of the plugin on the backend.
> 
>     * I got some errors related to the `security` client that the plugin adds under the `wazuh` route `context`. I could not find a solution.
> 
> 
> Unfortunately, these errors are the first ones, and we can't see any more errors until we solve them. We have to solve each error until we don't get any errors. After solving these problems, we have to test the UI and each functionality (monitoring, statistics, etc...)
> 
> I attach a patch that contains some fixes and workarounds to go over more errors as the first contact: [adapt_kibana_8.4.2.txt](https://github.com/wazuh/wazuh-kibana-app/files/9922002/adapt_kibana_8.4.2.txt)

Thank you for troubleshooting, it has been beneficial. I would like to mention that even after building the plugin from the source you provided, it is still encountering the same issue with version Kibana 8.4.2.

**Comment by Desvelao** [comment]
Hi @sqrtZeroKnowledge ,

Initially, I was working on fixing the compatibility problems with Kibana 8.4.2. I created a patch with some fixes, but there are more problems. Then, we moved to fix the compatibility problems with Kibana 8.5.0, but the issue was stopped and **it is currently on hold**. This means there are unsolved problems and we didn't test the plugin functionality.

We didn't try to build the plugin from the development branch we were working on, so it could have some problems we didn't identify.

What problem are you referring to?

**Comment by sqrtZeroKnowledge** [comment]
Thanks for your well detailed explanation,  The plug-in well installed for me but its not showing in the route uri of kibana /app/wazuh.

**Comment by Desvelao** [comment]
> Thanks for your well detailed explanation, The plug-in well installed for me but its not showing in the route uri of kibana /app/wazuh.

Hi @sqrtZeroKnowledge,

If you don't see the Wazuh plugin in the side menu of Kibana, this could mean there is a problem when trying to register the plugin. The Kibana logs could display any related information about that.

If the plugin appears in the side menu and when you access it, the page is blank, then this could mean there is some problem when mounting the application in the DOM.

Unfortunately, the current work done only fixes some problems (patch and development branch). If you need a stable plugin, then use an officially released plugin.

**Comment by gdiazlo** [comment]
Closing because we're going to use an integration. This will improve user experience for elastic users, adopting new elastic features quickly.  Related issue:
- https://github.com/wazuh/wazuh-kibana-app/issues/5182
